src/core_plugins/kibana/public/management/sections/indices/_edit.js

```diff
@@ -59,18 +59,22 @@ uiModules.get('apps/management')
   });
 
   $scope.refreshFilters = function () {
-    const indexedFieldTypes = $scope.indexPattern.fields
-      .filter(field => field.indexed)
-      .map(field => field.type);
-    const scriptedFieldLanguages = $scope.indexPattern.fields
-      .filter(field => field.scripted)
-      .map(field => field.lang);
+    const filterTypes = { indexedFieldTypes: [], scriptedFieldLanguages: [] };
+    const { indexedFieldTypes, scriptedFieldLanguages } = $scope.indexPattern.fields.reduce((acc, field) => {
+      if (field.indexed) {
```

Comment:
```
      // if (field.indexed) {
      //   acc.indexedFieldTypes.push(field.type);
      // }

      if (field.scripted) {
        acc.scriptedFieldLanguages.push(field.lang);
      } else {
        acc.indexedFieldTypes.push(field.type);
      }
```

https://github.com/elastic/kibana/blob/master/src/ui/public/index_patterns/_index_pattern.js#L251 
`index.getNonScriptedFields()` just checks `.scripted === false` which is the function that populates the indexed fields list. Less intuitive, but it avoids this situation:

![image](https://cloud.githubusercontent.com/assets/12193008/24167766/f289edae-0e45-11e7-8535-b692f1182f97.png)